### PR TITLE
SREP-2260 monitoring script for node health behind each available ELB

### PR DESCRIPTION
**Pull request by dofinn**

### CODE REVIEW

File: scripts/monitoring/cron-send-elb-health.py [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+    # Get a list of available ELBs
+    elbs_discovered=discover_elbs(elb_descriptions)
+
+    # Perform health check of each instance available behind each ELB
+    elb_health_check(client, elbs_discovered)
+
+    ### Metric Checks
+    if len(elb_no_instances) != 0:
+        for elb in range(len(elb_no_instances)):
+            print("ELB: %s has no instances behind it. Please investigate." % elb_no_instances[elb])
```
Comment (drewandersonnz) [inline scripts/monitoring/cron-send-elb-health.py:164]: I think this should increment `elb_instance_unhealthy_count_metric` 

Having no instances behind an ELB should be a fault condition.

File: scripts/monitoring/cron-send-elb-health.py [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+
+def main():
+    ''' Gather and examine details about this node within ELBs '''
+
+    args = parse_args()
+
+#    aws_access, aws_secret = get_aws_creds('~/.aws/credentials')
+    #instance_region = get_instance_region()
+    aws_access, aws_secret = get_aws_creds('./credentials')
+    instance_region = "eu-west-2"
```
Comment (drewandersonnz) [inline scripts/monitoring/cron-send-elb-health.py]: this shouldn't be hard-coded

File: scripts/monitoring/cron-send-elb-health.py [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+            elb_name = elbs_discovered[i]
+            elb_instance_name = elb_instance_response_state['InstanceId']
+            elb_instance_state = elb_instance_response_state['State']
+
+            # Check http response
+            if elb_response_http != 200:
+                print("A potential error occurred. HTTP Response: %s" % elb_response_http)
+
+            elb_instance_health(elb_instance_state, elb_instance_name, elb_name)
+#DEBUG            print("ELB: %s Instance: %s State: %s" % (elb_name, elb_instance_name, elb_instance_state))
```
Comment (drewandersonnz) [inline scripts/monitoring/cron-send-elb-health.py]: please remove commented code

File: scripts/monitoring/cron-send-elb-health.py [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+import yaml
+#import boto.ec2.elb
+#import boto.utils
+import boto3
+
+# number instances behind elb
+elb_no_instances = []
+
+# number of unhealthy instances
+elb_instance_unhealthy_count = []
```
Comment (bmeng) [inline scripts/monitoring/cron-send-elb-health.py]: The name of the variable is a little confusing, it names count, but it is actually an array. And a `len($var_name)` is a real count. Maybe we should rename this to something simple like `elb_unhealthy_instances`

**Comment by bmeng** [comment]
BTW, I think you should better submit the PR for `stg` branch firstly, and then `prod`.

**Review by drewandersonnz — APPROVED** [review]

**Pull request merged**